### PR TITLE
CMSIS v5 / RTOS v2 - on top of PAL R1.1 rel

### PR DESCRIPTION
**Pull request by JanneKiiskila**
This one is on top of the PAL R1.1 release, 1st (https://github.com/ARMmbed/pal/pull/60) pull was on top of the PAL R1.0 (which the [mbed-os-example-client](https://github.com/ARMmbed/mbed-os-example-client) is still using).

### CODE REVIEW

**Comment by avolinski** [comment]
pal 1.1 and lower should not support mbedOS5.5 or higher

File: Source/Port/Reference-Impl/mbedOS/RTOS/pal_plat_rtos.cpp [hunk 1]
```diff
@@ -78,64 +50,77 @@ typedef struct palThreadFuncWrapper{
...
+    mbed_rtos_storage_msg_queue_t osMessageQStorage;
 }palMessageQ_t;
 
 
 inline static void setDefaultThreadValues(palThread_t* thread)
 {
-#if PAL_UNIQUE_THREAD_PRIORITY      
-    g_palThreadPriorities[thread->osThread.tpriority+PRIORITY_INDEX_OFFSET] = false;
+#if PAL_UNIQUE_THREAD_PRIORITY
+    g_palThreadPriorities[thread->osThread.priority+PRIORITY_INDEX_OFFSET] = false;
```
Comment (JanneKiiskila) [inline Source/Port/Reference-Impl/mbedOS/RTOS/pal_plat_rtos.cpp]: You can add a commit on top of this, that sorts it if it's really bad.

File: Source/Port/Reference-Impl/mbedOS/RTOS/pal_plat_rtos.cpp [hunk 2]
```diff
@@ -21,54 +21,26 @@
...
-#include "cmsis_nvic.h"
-#include "core_cm7.h"
-#else
-// Declaration of NVIC_SystemReset is not present on core_ca9.h, but in the
-// board specific files. 
-void NVIC_SystemReset(void);
-#endif
-
-#include "critical.h"
+#include "mbed.h"
```
Comment (JanneKiiskila) [inline Source/Port/Reference-Impl/mbedOS/RTOS/pal_plat_rtos.cpp]: We don't need that compatibility - if someone wants to use the older one, they will have to keep the mbed-os.lib and pal.lib in sync. Anyone using mbed OS 5.5 or newer - use this, older ones use the PAL r1.1 from pal.lib (i.e. hash before this one).

The mbed-os-example-client keeps these in sync, so if you check out any version of the client -> you will get a working combination.

@yogpan01

**Comment by avolinski** [comment]
I will merge this PR in, And apply my changes on master later on.

**Pull request merged**